app/graphql/mutations/profile/associate_systems.rb

```diff
@@ -13,7 +13,7 @@ class AssociateSystems < BaseMutation
       def resolve(args = {})
         profile = find_profile(args[:id])
         add_inventory_hosts(args[:system_ids])
-        profile.update_hosts(args[:system_ids])
+        profile&.policy&.update_hosts(args[:system_ids])
```

Comment:
As a clarification, after this PR, profiles are only assigned hosts when a TestResult is uploaded.